ai-service/src/ai_service/embeddings/embedder.py

```
@@ -0,0 +1,138 @@
+
+    Raises:
+        EmbeddingError: If list of texts is empty, contains only empty strings, or encoding fails.
+    """
+    if not texts or all(not text.strip() for text in texts):
+        raise errors.EmbeddingError.empty_input()
+    model: SentenceTransformer = _get_model()
+
+    try:
+        # Use appropriate encoding method based on context
+        if is_query and hasattr(model, "encode_query"):
+            logger.debug(f"Encoding {len(texts)} queries using encode_query")
```

Ruff (0.12.2)

Logging statement uses f-string

(G004)